src/struct_info.rs

```diff
@@ -248,8 +248,12 @@ impl<'a> StructInfo<'a> {
             }
         });
         let mut target_generics = ty_generics.clone();
-        target_generics.push(syn::GenericArgument::Type(target_generics_tuple.into()));
-        ty_generics.push(syn::GenericArgument::Type(ty_generics_tuple.into()));
+
+        let ix = target_generics.iter().fold(0, |ix, arg| {
+            if let syn::GenericArgument::Type(_) = arg { ix } else { ix + 1 }
```

Comment:
BTW - why are you counting the arguments that **are not** `Type` instead of counting the arguments that **are** `Lifetime`? Wouldn't that mess up the index in presence of `Binding`/`Constraint`/`Const` generic parameters?

Reply:
My intention is to insert `ty_generics_tuple` right before original generic types.

In this way, I don't need to care if there is any generic arguments *before* or *after* the original generic types.

Reply:
Well, I don't count arguments that are **not** `Type`. I count the arguments **before** `Type`.

Reply:
I answered that in the other thread, but still - consider this:
```rust
struct Foo<'a, B: Ord, C> { /* ... */ }
```

With your approach, the builder state type will be placed between `B` (which is `Constraint`, not `Type`) and `C` (the first `Type`)

Reply:
I have played a little bit on the order of generic arguments.

Seems the allowed order is `Lifetime`, `Type`/`Constraint`, `Const`. (`Binding` is not allowed on a struct I suppose). 

It is faster and more straightforward to place state machine types after `Lifetime`.